### PR TITLE
Temporarily removing use of ReadOnlySpan indexer in Runtime.Extensions

### PR DESCRIPTION
**Pull request by ahsonkhan**
Removing the use of the ReadOnlySpan indexer to help the implementation change go through with the least disabling of tests.

This is a temporary workaround that is required to reduce the number of failing tests that would need to be disabled due to the change to ReadOnlySpan indexer to return ref readonly, here: https://github.com/dotnet/coreclr/pull/14727

cc @weshaggard, @stephentoub, @KrzysztofCwalina, @jkotas 

**This change should be reverted once the change propogates (and the tests enabled): https://github.com/dotnet/coreclr/issues/15089**

### CODE REVIEW

**Comment by stephentoub** [comment]
> Removing the use of the ReadOnlySpan indexer to help the implementation change go through with the least disabling of tests.

Why is this needed?  What breaks without it?

**Comment by jkotas** [comment]
> I don't like the idea of it being in the code base, even temporarily.

You can unsafe casts the ReadOnlySpan to Span to address this.

**Comment by jkotas** [comment]
So what is your plan to get this through? Which PR will got first, second, third, ... ?

**Comment by ahsonkhan** [comment]
> You can unsafe casts the ReadOnlySpan to Span to address this.

That is a good idea to avoid the performance hit. I will update it.

> So what is your plan to get this through? Which PR will got first, second, third, ... ?

1) This PR.
2) CoreCLR gets updated to point to the new version of the packages.
3) Followed by https://github.com/dotnet/coreclr/pull/14727
4) Then, we can merge this PR (when we get the coreclr update): https://github.com/dotnet/corefx/pull/24929

**Comment by jkotas** [comment]
And the plan for UWP NETNative?

File: src/System.Runtime.Extensions/src/System/IO/StreamReader.cs [hunk 1]
```diff
@@ -566,7 +567,13 @@ private bool IsPreamble()
                 return _checkPreamble;
             }
 
-            ReadOnlySpan<byte> preamble = _encoding.Preamble;
+            ReadOnlySpan<byte> readonlyPreamble = _encoding.Preamble;
+            Span<byte> preamble;
+            
+            unsafe
+            {
+                preamble = new Span<byte>(Unsafe.AsPointer<byte>(ref readonlyPreamble.DangerousGetPinnableReference()), readonlyPreamble.Length);
```
Comment (jkotas) [inline src/System.Runtime.Extensions/src/System/IO/StreamReader.cs]: This has GC hole...

Comment (jkotas) [reply src/System.Runtime.Extensions/src/System/IO/StreamReader.cs]: You need to pin the pointer while you are manipulating it.

**Comment by ahsonkhan** [comment]
> And the plan for UWP NETNative?

When step 3 happens (i.e. change to CoreCLR), I would make the change on the .NET Native side as well at the same time.

We can then merge the corefx PR (step 4), when we get the coreclr and .NET Native update.

Example of PRs that would need to be merged **together** with the corefx change (from step 4):
https://github.com/dotnet/corefx/pull/25362
https://github.com/dotnet/corefx/pull/25406

cc @davidwrighton, @tijoytom

**Pull request merged**